### ISSUE TITLE
Watch events : 405 (Method Not Allowed)

### ISSUE THREAD

**Issue by DTaste** [INFURA/infura#13]
Seems that watching events is no more allowed.
I'm trying to watch events from my contract this way :
` registerEvents.watch(function (err, result) { ...}`

It used to work few days ago and now get : 405 (Method Not Allowed) error
What happened ? 

Thanks ( and happy new year to the INFURA great team)

**Comment by egalano** [comment]
Hi @DTaste happy new year! Sorry for the inconvenience. We did rolled out a new version of our authentication API layer which filters out unstable/insecure RPC methods. We've been testing the feature on ropsten for several weeks and finally promoted it to mainnet. The registerEvents.watch function under the hood is attempting to install a filter onto one of the INFURA nodes. In the past we allowed this but the response behavior was non-deterministic. This is because we run multiple RPC nodes behind a load balancer so the filter would get installed on a specific node and if your subsequent getLogs requests hit a different node you would not get any events in your response payload even if there were actually events caught by the filter on the other RPC node on which the filter was actually installed.

We're working on finding a solution to this which can provide the functionality createFilter/getLogs was providing in a way which is scalable and secure. This feature is "Stateful Transaction Support" and is tracked in this issue #10

**Comment by DTaste** [comment]
I understand, thank you for your explanations.

**Comment by stefek99** [comment]
Wish I used search engine last night.

Spent some time debugging my code.

It was getting late, eventually gave up, went to sleep.

This morning realised what was the difference... Because I'm "read only", I switched from MetaMask _(that uses Infura under the hood, under the mask)_ to Infura directly. So minimalistic test case:

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Metamask vs Infura</title>

    <script>
    var address = "0x2328bc22d5705b2cde99e02a780e2fecdca4ad6b"; // https://ropsten.etherscan.io/address/0x2328bc22d5705b2cde99e02a780e2fecdca4ad6b
    var ABI = [{"constant":true,"inputs":[],"name":"timestampEnd","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"howManyGuaranteed","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"addr","type":"address"}],"name":"getPosition","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[],"name":"bid","outputs":[],"payable":true,"stateMutability":"payable","type":"function"},{"constant":true,"inputs":[],"name":"HEAD","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"initialPrice","outputs":[{"name":"","type":"bool"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"","type":"address"}],"name":"contributors","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"instructions","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"beneficiary","outputs":[{"name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"","type":"uint256"}],"name":"bids","outputs":[{"name":"prev","type":"uint256"},{"name":"next","type":"uint256"},{"name":"value","type":"uint256"},{"name":"contributor","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"getGuaranteedContributorsLenght","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[],"name":"finalize","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"name":"addr","type":"address"}],"name":"refundOnBehalf","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[],"name":"lastBidID","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[],"name":"refund","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[],"name":"howMany","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"","type":"address"}],"name":"bids","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"","type":"uint256"}],"name":"guaranteedContributors","outputs":[{"name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"increaseTimeIfBidBeforeEnd","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"description","outputs":[{"name":"","type":"string"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"getPosition","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"owner","outputs":[{"name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"name":"_description","type":"string"}],"name":"setDescription","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[],"name":"price","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"_contribution","type":"uint256"},{"name":"_startSearch","type":"uint256"}],"name":"searchInsertionPoint","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"LIMIT","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"finalized","outputs":[{"name":"","type":"bool"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"priceGuaranteed","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"getAccountListLenght","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"TAIL","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"increaseTimeBy","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"","type":"uint256"}],"name":"accountsList","outputs":[{"name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"winner","outputs":[{"name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"name":"","type":"address"}],"name":"guaranteedContributions","outputs":[{"name":"","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"name":"_instructions","type":"string"}],"name":"setInstructions","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"inputs":[{"name":"_price","type":"uint256"},{"name":"_description","type":"string"},{"name":"_timestampEnd","type":"uint256"},{"name":"_beneficiary","type":"address"},{"name":"_howMany","type":"uint256"},{"name":"_howManyGuaranteed","type":"uint256"},{"name":"_priceGuaranteed","type":"uint256"}],"payable":false,"stateMutability":"nonpayable","type":"constructor"},{"payable":true,"stateMutability":"payable","type":"fallback"},{"anonymous":false,"inputs":[{"indexed":false,"name":"addr","type":"address"},{"indexed":false,"name":"value","type":"uint256"},{"indexed":false,"name":"timestamp","type":"uint256"}],"name":"GuaranteedBid","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"number","type":"uint256"}],"name":"LogNumber","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"text","type":"string"}],"name":"LogText","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"addr","type":"address"}],"name":"LogAddress","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"bidder","type":"address"},{"indexed":true,"name":"price","type":"uint256"},{"indexed":true,"name":"timestamp","type":"uint256"}],"name":"BidEvent","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"addr","type":"address"},{"indexed":true,"name":"value","type":"uint256"},{"indexed":true,"name":"timestamp","type":"uint256"}],"name":"Refund","type":"event"}];

    // METAMASK WORKS / INFURA DOES NOT WORK
    web3 = new Web3(new Web3.providers.HttpProvider("https://ropsten.infura.io/hi8olE2lF8OqjyBSdtSm"));
    // web3 = new Web3(web3.currentProvider);

    var contract = web3.eth.contract(ABI).at(address);

    // CALLING REGULAR METHODS WORKS
    contract.getAccountListLenght(function(err, res) {
      console.log("How many accounts: " + res.toNumber());
    });

    // GETTING EVENTS DOES NOT WORK
    let bidEvent = contract.BidEvent({}, {fromBlock: 0, toBlock: 'latest'})
    bidEvent.get(function(err, events) {
      console.log(events);
    });
    </script>

  </head>
  <body>
  
  </body>
</html>
```

### Here is the error
![image](https://user-images.githubusercontent.com/249703/42732179-5778136e-8814-11e8-8f92-ef73a11e6a64.png)

---

1. Is there any workaround to get list of the events via Infura?
2. Can you please provide helpful error message? _(it's one search away - I found you - but `405` alone seems cryptic)_
3. If it works with MetaMask _(that uses Infura)_, it does not work with Infura directly - what to do how to live?

Talk to me in plain english please, 6 months ago didn't know how to deploy my own ERC20, now I'm working towards makign crypto usable for the wider audience. No need for Metamask to read data from the blockchain and this glitch alone cost me a few hours...

**Comment by rahul-ki** [comment]
HI is there any update on the issue? I am using UPort's web3 provider which is rinkeby.infura.io and getting the same error.

**Comment by wangw11** [comment]
HI is there any update on the issue? I am using UPort's web3 provider which is rinkeby.infura.io and getting the same error.

**Comment by reiven** [comment]
Hi, just came to ask about this issue. Is there any fix/workaround?
